VRDR/DeathRecord.cs

```diff
@@ -7388,6 +7758,1877 @@ public string EmergingIssue20
+                {
+                    foreach(Observation ob in RecordAxisCauseOfDeathObs)
+                    {
+                        CodeableConcept valueCC = (CodeableConcept)ob.Value;
+                        string value = valueCC.Coding[0].Code;
+                        Observation.ComponentComponent positionComp = ob.Component.Where(c => c.Code.Coding[0].Code=="position").FirstOrDefault();
+                        string position = positionComp.Value.ToString();
+                        Observation.ComponentComponent pregComp = ob.Component.Where(c => c.Code.Coding[0].Code=="wouldBeUnderlyingCauseOfDeathWithoutPregnancy").FirstOrDefault();
+                        string pregnancy = " ";
+                        if (pregComp != null && pregComp.Value != null)
+                        {
+                            string preg = pregComp.Value.ToString();
+                            pregnancy = preg == "true" ? "1" : " ";
+                            
+                        }
+                        Tuple<string, string, string> racod = Tuple.Create(position, value, pregnancy);
+
+                        rac.Add(racod);
+                    }
+                }
+                return rac.ToArray();
+            }
+            set
+            {
+                // clear all existing eac
+                Bundle.Entry.RemoveAll( entry => entry.Resource.ResourceType == ResourceType.Observation && (((Observation)entry.Resource).Code.Coding.First().Code == "80357-7"));
+                if (RecordAxisCauseOfDeathObs != null)
+                {
+                    RecordAxisCauseOfDeathObs.Clear();
+                }
+                else
+                {
+                    RecordAxisCauseOfDeathObs = new List<Observation>();
+                }
+                // Rebuild the list of observations
+                foreach(Tuple<string, string, string> rac in value)
+                {
+                    string position = rac.Item1;
+                    string val = rac.Item2;
+                    string preg = rac.Item3;
+
+                    Observation ob = new Observation();
+                    ob.Id = Guid.NewGuid().ToString();
+                    ob.Meta = new Meta();
+                    string[] recordAxis_profile = { ProfileURL.RecordAxisCauseOfDeath };
+                    ob.Meta.Profile = recordAxis_profile;
+                    ob.Code = new CodeableConcept(CodeSystems.LOINC, "80357-7", "Cause of death record axis code [Automated]", null);
+                    ob.Subject = new ResourceReference("urn:uuid:" + Decedent.Id);
+                    AddReferenceToComposition(ob.Id, "CodedContent");
+
+                    ob.Effective = new FhirDateTime();
+                    ob.Value = new CodeableConcept(CodeSystems.ICD10, val, null, null);
+
+                    Observation.ComponentComponent positionComp = new Observation.ComponentComponent();
+                    positionComp.Value = new FhirString(position);
```

Comment:
Same as above, position is an Integer not a FhirString

Reply:
updated